### ISSUE TITLE
TypeScript compilation error related to @types/express

### ISSUE THREAD

**Issue by JustinBeckwith** [googleapis/nodejs-logging-bunyan#243]
With a brand new module, that only has `@google-cloud/logging-bunyan` and `bunyan` installed, I get this error message:

```
> tsc -p .

node_modules/@google-cloud/logging/build/src/middleware/express/make-middleware.d.ts:16:49 - error TS7016: Could not find a declaration file for module 'express'. '/Users/beckwith/Code/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

16 import { Request, Response, NextFunction } from 'express';
                                                   ~~~~~~~~~
```

Installing `@types/express` made the bad message go away.  We should probably be distributing those types along with the package.

**Comment by ofrobots** [comment]
Similar issues have come up elsewhere in the past too. I am not sure what a good solution is. I am -1 on making `@types/express` a direct dependency of this module. Did you have a specific thing in mind when you said 'distribute along with the package'?

**Comment by ofrobots** [comment]
Ideally we need a concept of `promotableDevDependencies` in npm. The behavior would be that if you were to install devDependencies for a depend*ing* module, you promote the depend*ent*'s `promotable` dependencies and consider them to be the parent's `devDepencies`.

Let me illustrate.

* Lets say your package `A` depends on package `B` which depends on package `C`. Let's say `C` has a `promotableDevDependency` on `D`.
* As a developer of `A`, when you `npm install` in `A`:
  * npm installs `A`'s dependencies, i.e. `B`.
  * npm installs `B`'s dependencies, i.e. `C`.
  * npm installs `A`'s development dependencies.
  * npm DOES NOT install `B` or `C`'s devDependencies. (all of this up to now is existing npm behavior).
  * npm DOES NOT install `D` since `C` is not a direct dependency of `A`.
* As a developer of `B`, when you `npm install` in `B`:
  * npm installs `B`'s dependencies, i.e. `C`.
  * npm installs `B`'s development dependencies.
  * npm ALSO considers `promotableDevDependencies` of any of the direct dependencies to be a direct development dependency. That is, `D` also gets installed.

Another way of putting this would be, I want to be able to express the fact that if you are going to use `C` *during development*, then you need `D`.

Perhaps we can talk to npm about this behavior. I see this more and more with various typescript projects.

**Comment by ofrobots** [comment]
Fixed on googleapis/nodejs-logging#376.
